com.unity.sg2/Editor/GraphDeltaRegistry/Generation/Interpreter.cs

```diff
@@ -395,63 +424,48 @@ private static bool FunctionsAreEqual(ShaderFunction a, ShaderFunction b)
             return true;
         }
 
+        private static string ApplyConversionAndReturnConvertedVariable(PortHandler fromPort,
+                                                                        PortHandler toPort,
+                                                                        Registry registry,
+                                                                        ShaderContainer container,
+                                                                        ref ShaderFunction.Builder mainBodyFunctionBuilder,
+                                                                        string convertedTypeName,
+                                                                        ref ShaderFunctionRegistry shaderFunctions,
+                                                                        string toConvertNameOverride = null)
+        {
+            var connectedNode = fromPort.GetNode();
+            var toConvert = $"SYNTAX_{connectedNode.ID.LocalPath}_{fromPort.ID.LocalPath}";
+            if(toConvertNameOverride != null)
+            {
+                toConvert = toConvertNameOverride;
+            }
+            var converted = $"CONVERT_{connectedNode.ID.LocalPath}_{fromPort.ID.LocalPath}";
+            var cast = registry.GetCast(fromPort, toPort);
+            var castFunction = cast.GetShaderCast(fromPort.GetTypeField(), toPort.GetTypeField(), container, registry);
+
+            mainBodyFunctionBuilder.AddLine($"{convertedTypeName} {converted};");
+            mainBodyFunctionBuilder.AddLine($"{castFunction.Name}({toConvert}, {converted});");
+            shaderFunctions.EnsureFunctionPresent(castFunction);
+            return converted;
```

Comment:
446 should be:
`mainBodyFunctionBuilder.DeclareVariable(convertedType, converted);` (assuming you pass in the `ShaderType` instead of its name).

447 should be:
`mainBodyFunctionBuilder.CallFunction(castFunction, toConvert, converted);`

Josh nagged me the other day about using these instead of the janky param building I'd been doing up to this point.